### PR TITLE
Add EndpointSlice controller

### PR DESCRIPTION
**Pull request by alkar**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,9 +3,11 @@ module github.com/utilitywarehouse/kube-service-mirror
 go 1.14
 
 require (
+	github.com/google/go-cmp v0.3.0
 	github.com/hashicorp/go-hclog v0.11.0
 	github.com/stretchr/testify v1.4.0
 	k8s.io/api v0.17.4
 	k8s.io/apimachinery v0.17.4
 	k8s.io/client-go v0.17.4
+	k8s.io/utils v0.0.0-20191114184206-e782cd3c129f
```
Comment (ffilippopoulos) [inline go.mod:12]: can't this be same version as above, or a newer one, but not come from a commit?

File: runner.go [hunk 2]
```diff
@@ -321,31 +327,68 @@ func (r *Runner) updateEndpoints(name, namespace string, labels map[string]strin
...
+					"endpointslice", es,
+					"err", err,
+				)
+			}
+		} else {
+			log.Logger.Info(
+				"endpointslice found will try to update",
+				"endpointslice", name,
+			)
+			_, err := r.client.DiscoveryV1beta1().EndpointSlices(es.Namespace).Create(es)
```
Comment (ffilippopoulos) [inline runner.go:384]: is that `Update` then?

**Comment by ffilippopoulos** [comment]
Will an Endpoints `Modify` event cause anything on the related `Endpointslices`. From the code I cannot see that happening, but I do not know if there is any magic due to the owner ref that I didn't see (apart from the delete)

**Comment by alkar** [comment]
> Will an Endpoints `Modify` event cause anything on the related `Endpointslices`. From the code I cannot see that happening, but I do not know if there is any magic due to the owner ref that I didn't see (apart from the delete)

There's no magic! Only for deletion. Just a stashed changed that was never committed)

File: go.mod (hunk 1 above)
Comment (alkar) [reply go.mod:12]: This was added automatically, probably by `go mod tidy` magic. Do you want to declare a version manually?

Comment (ffilippopoulos) [reply go.mod:12]: I think that it should match the above `k8s.io` submodules

Comment (alkar) [reply go.mod:12]: That package doesn't really have releases https://github.com/kubernetes/utils/releases

**Pull request closed without merging**